lib/codecs/tests/data/native_encoding/schema.cue

```diff
@@ -0,0 +1,46 @@
+#Event: {log: #Log} | {metric: #Metric} | {trace: #Trace}
+
+#Log: {...}
+
+#Trace: {...}
+
+#Metric: {
+	name:       string
+	namespace?: string
+	tags?: {[string]: string}
+	timestamp?: #Timestamp
+	kind:       "incremental" | "absolute"
+	{counter: value: number} |
+	{gauge: value: number} |
+	{set: values: [...string]} |
+	{distribution: {
+		samples: [...{value: number, rate: int}]
+		statistic: "histogram" | "summary"
+	}
+	} |
+	{aggregated_histogram: {
+		buckets: [...{upper_limit: number, count: int}]
+		count: int
+		sum:   number
+	}} |
+	{aggregated_summary: {
+		quantiles: [...{quantile: number, value: number}]
+		count: int
+		sum:   number
+	}} |
+	{sketch:
+		sketch: AgentDDSketch: {
+			bins: {
```

Comment:
This bit seems a little odd, I assume because we just never had a reason to think about the JSON representation of these structs. Do we think it's worth flattening these out to just `{sketch: bins: { ...` or similar?

Reply:
cc/ @tobz and @prognant for thoughts here.

Reply:
Noting that we discussed this the other day and decided to just proceed with it as-is.